src/components/DatePicker/CalendarHeader/CalendarHeader.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { ReactDatePickerCustomHeaderProps } from 'react-datepicker';
+import { ImArrowLeft2, ImArrowRight2 } from 'react-icons/im';
+import { bem } from '../../../utils';
+import { Button } from '../../Buttons';
+import { Text } from '../../Text';
+import styles from './CalendarHeader.scss';
+
+interface Props extends ReactDatePickerCustomHeaderProps {
+    /** the locale in which the names of the month should be displayed. Format should match Intl definitions */
+    locale?: string;
+    /** the min and max year selectable by the user */
+    yearsRange: [number, number];
+}
+
+const { block, elem } = bem('CalendarHeader', styles);
+
+export const CalendarHeader: React.FC<Props> = (props) => {
+    const {
+        locale = 'en',
+        yearsRange,
+        date,
+        decreaseMonth,
+        prevMonthButtonDisabled,
+        increaseMonth,
+        nextMonthButtonDisabled,
+        changeYear,
+    } = props;
+
+    const monthName = new Intl.DateTimeFormat(locale, { month: 'long' }).format(date);
+
+    const [minYear, maxYear] = yearsRange;
```

Comment:
Let's add a warning if `minYear` and `maxYear` are in the wrong condition. For instance, `minYear` is more than  `maxYear`. In this scenario we have the error: 
![Screenshot 2021-12-13 at 12 03 23](https://user-images.githubusercontent.com/19436304/145801936-78ec7816-09e5-4ea7-8bac-3cb03d6b4d52.png)